src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/CompletionHandler.cs

```diff
@@ -153,27 +148,25 @@ internal class CompletionHandler : IRequestHandler<CompletionParams, SumType<Com
             {
                 // Set some context on the CompletionItem so the CompletionResolveHandler can handle it accordingly.
                 result = SetResolveData(result.Value, serverKind);
-                if (IsSimpleImplicitExpression(documentSnapshot, request, serverKind))
+
+                var wordExtent = GetWordExtent(documentSnapshot, request, serverKind);
+                if (IsSimpleImplicitExpression(documentSnapshot, request, wordExtent, serverKind))
                 {
                     result = DoNotPreselect(result.Value);
-                    result = IncludeCSharpKeywords(result.Value, serverKind);
+                    result = IncludeCSharpKeywords(result.Value);
                 }
+
+                result = RemoveDesignTimeItems(result.Value, documentSnapshot, wordExtent, serverKind);
             }
 
             return result;
         }
 
-        private bool IsSimpleImplicitExpression(LSPDocumentSnapshot documentSnapshot, CompletionParams request, LanguageServerKind serverKind)
+        private TextExtent? GetWordExtent(LSPDocumentSnapshot documentSnapshot, CompletionParams request, LanguageServerKind serverKind)
         {
             if (serverKind != LanguageServerKind.CSharp)
```

Comment:
It looks like we're doing this serverkind C# check in several locations now. Lets wrap the IncludeCSharpKeywords bits and the RemoveDesignTimeItems bit in the check, that way each method doesn't need to recheck the same thing. Could even refactor all the C# massaging logic into its own method since it's conceptually a different stage of completion